src/generic/TotalFraction.jl

```diff
@@ -0,0 +1,613 @@
+###############################################################################
+#
+#   Fraction.jl : fraction fields
+#
+###############################################################################
+
+export TotalRingOfFractions
+
+###############################################################################
+#
+#   Data type and parent object methods
+#
+###############################################################################
+
+parent_type(::Type{TotFrac{T}}) where T <: RingElem = TotFracRing{T}
+
+elem_type(::Type{TotFracRing{T}}) where {T <: RingElem} = TotFrac{T}
+
+###############################################################################
+#
+#   Data type and parent object methods
+#
+###############################################################################
+
+base_ring(a::TotFracRing{T}) where T <: RingElem = a.base_ring::parent_type(T)
+
+base_ring(a::TotFrac) = base_ring(parent(a))
+
+parent(a::TotFrac) = a.parent
+
+function is_domain_type(::Type{T}) where {S <: RingElement, T <: TotFrac{S}}
+   return is_domain_type(S)
+end
+
+function is_exact_type(a::Type{T}) where {S <: RingElement, T <: TotFrac{S}}
+   return is_exact_type(S)
+end
+
+function characteristic(R::TotFracRing{T}) where T <: RingElem
+   return characteristic(base_ring(R))
+end
+
+function check_parent(a::TotFrac, b::TotFrac, throw::Bool = true)
+   fl = parent(a) != parent(b)
+   fl && throw && error("Incompatible rings in total ring of fractions operation")
+   return !fl
+end
+
+###############################################################################
+#
+#   Constructors
+#
+###############################################################################
+
+function //(x::T, y::TotFrac{T}) where {T <: RingElem}
+   parent(y)(x*denominator(y, false), numerator(y, false))
+end
+
+function //(x::TotFrac{T}, y::T) where {T <: RingElem}
+   parent(x)(numerator(x, false), denominator(x, false)*y)
+end
+
+function //(x::Integer, y::TotFrac{T}) where {T <: RingElem}
+   parent(y)(x*denominator(y, false), numerator(y, false))
+end
+
+function //(x::TotFrac{T}, y::Integer) where {T <: RingElem}
+   parent(x)(numerator(x, false), denominator(x, false)*y)
+end
+
+###############################################################################
+#
+#   Basic manipulation
+#
+###############################################################################
+
+function Base.numerator(a::TotFrac, canonicalise::Bool=true)
+   return a.num
+end
+
+function Base.denominator(a::TotFrac, canonicalise::Bool=true)
+   return a.den
+end
+
+function deepcopy_internal(a::TotFrac{T}, dict::IdDict) where {T <: RingElem}
+   v = TotFrac{T}(deepcopy(numerator(a, false)), deepcopy(denominator(a, false)))
```

Comment:
I think these deepcopys should pass the iddict along

Reply:
Ok. I also fixed the same for FractionField as it was done the same way there.

Reply:
pass along via another deepcopy_internal

Reply:
ok, fixed.